packages/fx-core/src/component/driver/aad/create.ts

```diff
@@ -217,12 +221,18 @@ export class CreateAadAppDriver implements StepDriver {
           if (
             error.response!.status === 403 &&
             message.includes(constants.insufficientPermissionErrorMessage) &&
-            !isTestToolEnabledProject(context.projectPath)
+            !isTestToolEnabledProject(context.projectPath) &&
+            process.env.TEAMSFX_ENV == environmentNameManager.getLocalEnvName()
```

Comment:
minor: why only enable the new feature for local environment now?

Reply:
Because this feature is targeting for F5 success rate, so I'd like to limit the impact.